sample-app/src/main/java/kin/devplatform/sample/MainActivity.java

```diff
@@ -138,20 +140,37 @@ private void startSdk() {
 		 * */
 		String jwt = SignInRepo.getJWT(this);
 		Kin.enableLogs(true);
-		Kin.start(getApplicationContext(), jwt, Environment.getPlayground(), new KinCallback<Void>() {
+		Kin.start(getApplicationContext(), "test", jwt, Environment.getPlayground(), new KinCallback<Void>() {
 			@Override
 			public void onResponse(Void response) {
+				// TODO: 09/01/2019 maybe remove the progress bar that we added when migration was started?
 				Toast.makeText(MainActivity.this, "Starting SDK succeeded", Toast.LENGTH_LONG).show();
 				addNativeSpendOffer(nativeSpendOffer);
 				addNativeOfferClickedObserver();
-
+				addBalanceObserver();
+				isInitialized = true;
 			}
 
 			@Override
 			public void onFailure(KinEcosystemException error) {
 				Toast.makeText(MainActivity.this, "Starting SDK failed", Toast.LENGTH_LONG).show();
 				Log.d(TAG, "Kin.start() failed with =  " + ErrorUtil.getPrintableStackTrace(error));
 			}
+		}, new KinMigrationListener() {
+			@Override
+			public void onStart() {
+				// TODO: 09/01/2019 use some progress bar.
```

Comment:
We should have an UI for all of this events for testing purposes

Reply:
Of course, we will add it as soon as possible after will finish with the integration of migration module into the devplatform